### PR TITLE
vending machine funtimes 2

### PR DESCRIPTION
**Pull request by ghost**
### Vending machine changes

Changes the vending machine wire that toggles it firing at people, now it just sets the firing to off.

Reason for this is so that vending machines aren't so easily cheesed

lotsofstock

Roughly $15000 per vending machine, that can be acquired with a multitool, a screwdriver, and throwing a mouse in the room and waiting.

So, now the only way for a vending machine to fire its products at others, is if it goes rogue, the second part of this PR

### Brand intelligence event renewal

Rather than it being almost an entirely beneficial event, now any vending machine affected by this viral marketing may end up becoming a mimic-like entity, throwing itself at would-be patrons to force them to buy their product. Giving them some money earns you their friendship for a certain amount of time, depending on the amount of money you have given them.

The way to stop this from happening is to silence or otherwise break the controlling vending machine, which now talks a lot more often, as well as could potentially be given away by a news report regarding instances of the vending machine (Possibility of this news report is increased with 'well connected' stations that would have a librarian or a reporter)

Previously the controlling vending machine would only talk every 120 ticks (Being within a multipleOf 12 check within a multipleOf 5 check) so now it's a little more obvious which vending machine is in control. Said controlling vending machine also announces which vending machine under its control it is going to go for more 'direct' trading.

### Additional non-atomic changes

To help with testing this, I added a new thing to the admins secret window. A button that allows you to choose from the list of possible random events to trigger.

:cl:
* tweak: Vending machine firing wire now only turns the firing mechanism off
* tweak: Rampant vending machine event tweaked to be a bit more dangerous and engaging

### CODE REVIEW

**Comment by ghost** [comment]
Mining vendor being converted to a vending machine was a mistake

**Comment by jknpj** [comment]
poorfags on suicide watch

**Comment by yclatious** [comment]
Personally, I find it stupid how you want to gut assistant dens with tons of food being made from vendors just because cargo gets a ton of junk it almost never actually uses out of a machine that has maybe three items you actually purchase all the time with the rest being gimmicky or just a small boost that usually gets wasted or not used as well as people think it does.

Really, I never needed to buy more than some soap for when cargo decided looking like a greyshit slum was okay, a PK for goliaths that got too close and some chilli to deal with basilisks that were in complicated spots, the pokemon set is just for gimmicky fun/treason and some items are just useless like the canary, the facehugging shittoy someone decided to code in for some reason and unpractical garbage like the wormhole jaunter that throws you in who knows where with a free puke stun to boot, on some days I did decide to go for the pokemon gimmick and farmed goliaths and all but the difference in availability from the bought and the hacking is somewhat minimal anyhow, you wont wait five minutes to actually strike the earth so you get a lucky jackhammer, youl either ask the QM for one or you will just mine and a large amount of the time, when you do come back, the "loot piles" the cargo machines made from the throwing wire were usually composed of about twelve soap bars for mass slipping, a ton of shards from the whiskey bottles, shitty facehugger toys, a couple random teleportation puking devices, 4 chilly, a few pokeballs, maybe a single advanced lazarus, 3 knife holsters, stuff you can just buy easy with your first back money anyhow, then again maybe I just got unlucky on my shifts where the techs hacked it?

Not like the machine even has anything really impressive in it, the jetpack is okay but its not gamechanging by any means, a great tool for vault hunting but I never really saw it being used too much or too many bought from the machine/saw any in the loot piles, the drone is alright but using PKs and going monster hunting is just more fun, like I said before the jaunt thing is usually shit as its fully random where you go and a certain tell you were there from the puke and if you land in sec its a 50/50 if sec will chill or arrest your ass, by the time you get back you should already have diamonds so the jackhammer is useless unless science is fully gone, pokemon stuff is just for fun/traitoring and usually the traitor miner belt is better for antagging unless its round end shuttle grief, so I really don't see the point in needing to gut all vending machines and the vox/greyshit food lair classic just to stop what Ive seen to be just shitty piles of never well utilized loot anyways since cargo is a newbie pit stop and almost nobody who plays cargo can fully utilize it well, just snowflake the mining machine if you really have a issue with it shooting items but it sounds a lot more like you hate it for the inflated shitconomy number than you hate it by the actual application and effect on the round, because in real rounds ive never seen the loot dropped be good enough to actually care or seen the loot dropped be good enough to alter the survivability of cargo as a department anyhow.

TL:DR, just snowflake it so you don't gut the fun assistant dens, the loot piles often dont do dick and cant stop cargo from being idiots and when you come back from mining you can buy anything you want anyhow. Keep the event though, sounds like !FUN!

**Comment by RealestEstate** [comment]
damn, this means no more bola firing ghetto turrets being made out of sec vends and custom vending machines. can't we just make the paid items unexploitable via firing?

**Comment by ghost** [comment]
My brain says haha no more miner scumming but my heart says bye bye ghetto chem shops. What a terrible curse.

**Comment by Chazach** [comment]
The miners can buy the whole fucking world after 5 minutes on the roid, all this does is kill off !!!FUN!!! and prevent massive dens of foodstuffs

**Comment by ghost** [comment]
Then blow the fucking thing up with an IED

`/obj/machinery/vending/ex_act(severity)
	switch(severity)
		if(1.0)
			qdel(src)
			return
		if(2.0)
			if (prob(50))
				qdel(src)
				return
		if(3.0)
			if(prob(25))
				malfunction()`

or EMP it

`/obj/machinery/vending/emp_act(severity)
	if(stat & (BROKEN|NOPOWER))
		return
	switch(severity)
		if(1.0)
			malfunction()
		if(2.0)
			if(prob(50))
				malfunction()
		if(3.0)
			if(prob(25))
				malfunction()`

Or just hit the blumin thing

`/obj/machinery/vending/proc/damaged()
	src.health -= 4
	if(src.health <= 0)
		stat |= BROKEN
		src.update_vicon()
		return
	if(prob(2)) //Jackpot!
		malfunction()
	if(prob(2))
		src.TurnOff(600) //A whole minute`

Or go sentient machine hunting in the new event. Put some blumin effort into it

**Comment by yclatious** [comment]
Hold up, you saying that instead of doing the logical thing and just fixing the issue with the one machine you dislike the function so that you can keep a fun feature it that is cherished by greyshits like me and generaly liked, you still fuck all the machines up, nerf them hard, remove several fun gimmicks and fun to be had with several different types of machines and excuse it on three methods that are worse, one of which a average greyshit cant even do, or the event?

Dragonbro, thats just a load of shit.

**Comment by Exxion** [comment]
Vending machines shooting stuff has always been awful and obnoxious

**Comment by N3X15** [comment]
Non-atomic changes.
